sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -293,19 +293,23 @@ trait JoinSelectionHelper extends Logging {
       join: Join,
       hintOnly: Boolean,
       conf: SQLConf): Option[BuildSide] = {
-    val buildLeft = if (hintOnly) {
-      hintToBroadcastLeft(join.hint)
-    } else {
-      canBroadcastBySize(join.left, conf) && !hintToNotBroadcastLeft(join.hint)
+    def maybeBuildLeft(): Boolean = {
```

Comment:
Is it ok to just change `val` to `def`? This way, we can keep the changes to a minimum. WDYT? @cloud-fan @yaooqinn @beliefer

Reply:
`buildLeft` seems wired if it used as function. It seems we build something here.

Reply:
how about `def shouldBuildLeft`?